AirCasting/SessionViews/SessionCartViewModel/MeasurementsViewModel.swift

```diff
@@ -0,0 +1,60 @@
+// Created by Lunar on 11/09/2021.
+//
+
+import Foundation
+import CoreLocation
+import CoreData
+
+class ViewModel: NSObject, ObservableObject {
+    
+    @Published var showLoadingIndicator = false
+    let sessionDownloader = SessionDownloadService(client: URLSession.shared,
```

Comment:
instantiating own dependencies is not a good design, it should be passed in from the outside (probably using init). What's more - it's not protocolozed and as it's boundary-crossing (`sessionDownloader` is in delivery departament and viewmodels are UI) and it really should be. We can create a protocol for it real quick and easy:

```Swift
protocol MeasurementsDownloadable {
    func downloadSessionWithMeasurement(uuid: SessionUUID, completion: @escaping (Result<SessionsSynchronization.SessionDownstreamData, Error>) -> Void) -> Cancellable
}
```

and than just conform SessionDownloadService to it (as it's already doing so):

```Swift
final class SessionDownloadService: SessionDownstream, MeasurementsDownloadable {
...
}
```

This way here we can just add a dependency over this protocol instead of a whole `SessionDownloadService`:

```Swift
private let sessionDownloader: MeasurementsDownloadable
...
init(sessionDownloader: MeasurementsDownloadable ... ) { ... }
```